framework/include/controls/ControlInterface.h

```diff
@@ -129,98 +86,59 @@ class ControlInterface
   const std::string & _ci_name;
 
   // Only these objects are allowed to construct this interface
-  friend class RealParameterReporter;
+  friend class RealControlParameterReporter;
   friend class Control;
 };
 
 template<typename T>
-T &
+ControllableParameter<T>
 ControlInterface::getControlParam(const std::string & name)
 {
-  return getControlParamByName<T>(_ci_parameters.get<std::string>(name));
+  return getControlParamByName<T>(_ci_parameters.get<std::string>(name),
+                                  std::numeric_limits<unsigned int>::max());
 }
 
 template<typename T>
-T &
-ControlInterface::getControlParamByName(const std::string & name)
+ControllableParameter<T>
+ControlInterface::getControlParam(const std::string & name, unsigned int count)
```

Comment:
The user could feed an input parameter to the count so it could be set from the input file. I just fear blindly changing parameters, perhaps we need a debug flag or something that would alert you what parameters are changing and when.